examples/oidc/public-vc/react/src/services/keycloak.js

```diff
@@ -0,0 +1,72 @@
+import Keycloak from 'keycloak-js';
+
+const _kc = new Keycloak({
+  url: process.env.REACT_APP_SSO_AUTH_SERVER_URL,
+  realm: process.env.REACT_APP_SSO_REALM,
+  clientId: process.env.REACT_APP_SSO_CLIENT_ID,
+});
+
+const loginOptions = {
+  redirectUri: process.env.REACT_APP_SSO_REDIRECT_URI,
+  idpHint: '',
+  pres_req_conf_id: process.env.REACT_APP_PRES_REQ_CONF_ID,
+};
+
+export const initializeKeycloak = async () => {
+  try {
+    _kc.onTokenExpired = () => {
+      _kc
+        .updateToken(5)
+        .then(function (refreshed) {
+          if (refreshed) {
+            alert('Token was successfully refreshed');
+          } else {
+            alert('Token is still valid');
+          }
+        })
+        .catch(function () {
+          alert('Failed to refresh the token, or the session has expired');
+        });
+    };
+
+    const auth = await _kc.init({
+      pkceMethod: 'S256',
+      checkLoginIframe: false,
+      onLoad: 'check-sso',
+    });
+
+    if (auth) {
+      console.log("Got here?")
+      return _kc;
+    } else {
+      console.log(loginOptions)
+      if(loginOptions.pres_req_conf_id){
+        var loginURL = _kc?.createLoginUrl(loginOptions);
+        console.log(loginURL)
```

Comment:
This one too probably, might be good to keep the loginOptions log though with some descriptor text since this is an example app